src/Domain/DomainHelpers.hpp

```diff
@@ -0,0 +1,638 @@
+                   *(maps_of_all_blocks[block1_index]),
+                   *(maps_of_all_blocks[block2_index]),
+                   corners_of_all_blocks[block1_index],
+                   corners_of_all_blocks[block2_index]) < tolerance,
+               "Embeddingmaps "
+                   << block1_index << " and " << block2_index
+                   << " were found to share a boundary according to the "
+                      "corner numbering scheme applied. However, "
+                      "they do not share this boundary in "
+                      "physical space within a tolerance of "
+                   << tolerance << ". The PhysicalSeparation between them is:"
+                   << DomainHelpers<VolumeDim>::physical_rms_separation(
+                          *(maps_of_all_blocks[block1_index]),
+                          *(maps_of_all_blocks[block2_index]),
+                          corners_of_all_blocks[block1_index],
+                          corners_of_all_blocks[block2_index]));
+        auto orientation_helper =
+            DomainHelpers<VolumeDim>::obtain_correspondence_between_blocks(
+                block1_index, block2_index, corners_of_all_blocks);
+        neighbor.emplace(std::move((orientation_helper.first)[0]),
+                         BlockNeighbor<VolumeDim>(
+                             block2_index, Orientation<VolumeDim>(
+                                               orientation_helper.first,
+                                               orientation_helper.second)));
+      }
+    }
+    neighbors_of_all_blocks.push_back(neighbor);
+  }
+}
+
+template <size_t VolumeDim>
+void DomainHelpers<VolumeDim>::set_periodic_like_boundaries(
+    const std::vector<std::array<size_t, two_to_the(VolumeDim - 1)>>&
+        identifications,
+    const std::vector<std::array<size_t, two_to_the(VolumeDim)>>&
+        corners_of_all_blocks,
+    std::vector<
+        std::unordered_map<Direction<VolumeDim>, BlockNeighbor<VolumeDim>>>&
+        neighbors_of_all_blocks) {
+  ASSERT(identifications.size() % 2 == 0,
+         "You must specify an even number of identifications.");
+  for (size_t i = 0; i < identifications.size() / 2; i++) {
+    const auto& face1 = identifications[2 * i];
+    const auto& face2 = identifications[2 * i + 1];
+    size_t id = DomainHelpers<VolumeDim>::find_block_id(face1, face2,
+                                                        corners_of_all_blocks);
+    auto face1_canon = DomainHelpers<VolumeDim>::get_common_local_corners(
+        corners_of_all_blocks[id], face1);
+    auto face2_canon = DomainHelpers<VolumeDim>::get_common_local_corners(
+        corners_of_all_blocks[id], face2);
+    auto face1_normal_dir =
+        DomainHelpers<VolumeDim>::get_direction_normal_to_face(
+            DomainHelpers<VolumeDim>::logical_coords_of_corners(face1_canon));
+    auto face2_normal_dir =
+        DomainHelpers<VolumeDim>::get_direction_normal_to_face(
```

Comment:
(Choosing an arbitrary representative line) You are inconsistent throughout the file with whether the functions are qualified with the class name or not.  I'd suggest removing the unneccesary ones to avoid clutter.